### DISCUSSION TITLE
Wrapping useFetch loses types

### DISCUSSION THREAD

**Discussion by lacorde** [nuxt/nuxt#22702] [Questions]
Let's say I have the simple following `server/api/person.get.ts` file :

```
interface Person {
  name: string,
  age: number
}

export default defineEventHandler(async (event) => {
  const person: Person = {
    name: 'John',
    age: 29
  }
  return person;
});
```
In my app.vue, I have the following call :

`const { data } = useFetch('/api/person');`

After that, hint typing `data.value` suggests me `name` and `age`, which is great.

But if I were to wrap useFetch in a composable (for adding default headers for example)
```
export const useCustomFetch = async (request: NitroFetchRequest) => {
  return await useFetch(request);
};
```
I lose all of this autocompletion ! Any way to achieve this ?

**Comment by WailanTirajoh** [comment]
https://github.com/nuxt/nuxt/discussions/22532

This maybe related with your question.
